src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -180,7 +180,7 @@ public function searchBarAction(Request $request)
             $this->cspHandler->disableCsp();
         }
 
-        if (null === $session = $request->getSession()) {
+        if (!$request->hasSession() || !$session = $request->getSession()) {
```

Comment:
`if (!$request->hasSession())` is enough

Reply:
Are you sure? I think we need $session in the else part.
We could move `$session = $request->getSession()` into the else though.

Reply:
moving it to the `else` seems reasonable to me